src/index.ts

```diff
@@ -1,61 +1,126 @@
-import guid from "./utils/guid.js";
-import errors from "./errors/index.js";
-import Live from "./classes/live.js";
-import Socket from "./classes/socket.js";
-import Pinger from "./classes/pinger.js";
-import Emitter from "./classes/emitter.js";
+// deno-lint-ignore-file no-explicit-any
+import guid from "./utils/guid.ts";
+import errors from "./errors/index.ts";
+import Live from "./classes/live.ts";
+import Socket from "./classes/socket.ts";
+import Pinger from "./classes/pinger.ts";
+import Emitter from "./classes/emitter.ts";
+
+let singleton: Surreal;
+
+interface BasePatch {
+	path: string;
+}
+
+export interface AddPatch extends BasePatch {
+  op: 'add';
+  value: any;
+}
+
+export interface RemovePatch extends BasePatch {
+	op: 'remove';
+}
+
+export interface ReplacePatch extends BasePatch {
+	op: 'replace';
+  value: any;
```

Comment:
I think some formatting tools could help to avoid inconsistencies in indentation and such.
Maybe adding eslint & prettier to the library makes sense?

Reply:
I would say useing `deno fmt` would be more fitting (but totaly agree!)